app/controllers/account_reset/delete_account_controller.rb

```diff
@@ -0,0 +1,48 @@
+module AccountReset
+  class DeleteAccountController < ApplicationController
+    before_action :check_feature_enabled
+    before_action :prevent_parameter_leak, only: :show
+    before_action :check_granted_token
+
+    def show; end
+
+    def delete
+      analytics.track_event(Analytics::ACCOUNT_RESET, event: :delete, token_valid: true)
+      email = reset_session_and_set_email
+      UserMailer.account_reset_complete(email).deliver_later
+      redirect_to account_reset_confirm_delete_account_url
+    end
+
+    private
+
+    def check_feature_enabled
+      redirect_to root_url unless FeatureManagement.account_reset_enabled?
+    end
+
+    def reset_session_and_set_email
+      user = @account_reset_request.user
+      email = user.email
+      user.destroy!
+      sign_out
+      session[:email] = email
```

Comment:
If we use the `flash`, then this would be the other place we would touch (besides the two controllers above).